core/src/view/view.cpp

```diff
@@ -188,6 +188,8 @@ void View::screenToGroundPlane(float& _screenX, float& _screenY) const {
     
     _screenX = ray_world.x;
     _screenY = ray_world.y;
+    
+    return glm::vec2(t, ray_world.z);
```

Comment:
Do we need both `t` and `z`? Isn't `t < 0` sufficient to exclude backwards intersections?